guides/release/components-update/component-state.md

```diff
@@ -0,0 +1,306 @@
+While you can accomplish a lot in Ember using HTML templating, you'll need JavaScript to make your application interactive.
+
+Let's start with a really simple example, a counter component. When the user presses the `+` button, the count will increase by 1. When the user presses the `-` button, the count will decrease by 1.
+
+First, let's start with the HTML.
+
+```handlebars {data-filename="app/components/counter.hbs"}
+<p>0</p>
+
+<button>+1</button>
+<button>-1</button>
+```
+
+## Tracked Properties
+
+To make this work, we will need to stop hardcoding the number, and we will need to wire up the buttons.
+
+```js {data-filename="app/components/counter.js"}
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+
+export default class extends Component {
+  @tracked count = 0;
+}
+```
+
+There's a bunch going on here, but the most important part is `@tracked count = 0`. This creates a dynamic value that you can stick in the template, instead of hardcoding it.
+
+```handlebars {data-filename="app/components/counter.hbs" data-diff="-1,+2"}
+<p>0</p>
+<p>{{this.count}}</p>
+
+<button>+1</button>
+<button>-1</button>
+```
+
+When we say `this.count` in the component template, we're referring to a property that we defined in the JavaScript class.
+
+The output looks the same as before, but now the `0` comes from JavaScript, and we can change it from the button.
+
+## HTML Modifiers and Actions
+
+Next, we want to wire up the buttons. When the user presses `+1`, we want `this.count` to go up by 1. When the user presses `-1`, we want it to go down by 1.
+
+To attach an event handler to an HTML tag, you use the `on` HTML modifier. HTML modifiers are an Ember syntax that allow you to attach logic to a tag.
```

Comment:
Have we defined event handlers? This seems to be the first place we introduce them, but it feels a little random. Should we also italicize this?